src/fidesops/ops/service/saas_request/override_implementations/firebase_auth_request_overrides.py

```diff
@@ -0,0 +1,155 @@
+import logging
+from typing import Any, Dict, List
+
+import firebase_admin
+from firebase_admin import App, auth, credentials
+from firebase_admin.auth import UserRecord
+
+from fidesops.ops.graph.traversal import TraversalNode
+from fidesops.ops.models.policy import Policy
+from fidesops.ops.models.privacy_request import PrivacyRequest
+from fidesops.ops.service.saas_request.saas_request_override_factory import (
+    SaaSRequestType,
+    register,
+)
+from fidesops.ops.util.collection_util import Row
+
+logger = logging.getLogger(__name__)
+
+
+@register("firebase_auth_user_access", [SaaSRequestType.READ])
+def firebase_auth_user_access(
+    node: TraversalNode,
+    policy: Policy,
+    privacy_request: PrivacyRequest,
+    input_data: Dict[str, List[Any]],
+    secrets: Dict[str, Any],
+) -> List[Row]:
+    """
+    SaaS Request Override function used to integrate access requests with
+    Firebase Auth service. Specifically, it reads `User` records.
+    """
+    app = initialize_firebase(secrets)
+
+    processed_data = []
+
+    emails = input_data.get("email", [])
+    for email in emails:
+        user: UserRecord = auth.get_user_by_email(email, app=app)
+        row = {"email": user.email, "uid": user.uid}
+        if user.display_name is not None:
+            row["display_name"] = user.display_name
+        if user.phone_number is not None:
+            row["phone_number"] = user.phone_number
+        if user.disabled is not None:
+            row["disabled"] = user.disabled
+        if user.email_verified is not None:
+            row["email_verified"] = user.email_verified
+        if user.provider_data:
```

Comment:
i was having a bit of trouble testing this functionality because i don't think we're able to programmatically create provider data on a given user, so we can't seed the test data for this. i'll look to see if i can manually verify the functionality moving forward.

Reply:
interesting, thanks for digging that up! i'm with you on keeping the tests self-contained (and trying to run a CLI command within a pytest harness doesn't sound very appealing), but evaluating the provider data functionality may end up being important...so i'm kinda split here. i'll try to do some more digging on my end, thanks for the pointer.

Reply:
noting that i found an SDK [function](https://firebase.google.com/docs/reference/admin/python/firebase_admin.auth?authuser=2#import_users)/[class](https://firebase.google.com/docs/reference/admin/python/firebase_admin.auth?authuser=2#importuserrecord) that allows us to import users with provider data defined. i'm updating test fixtures to create our sample users with provider data and include those pieces of data in the tests.